service/history/historyEngine.go

```diff
@@ -1289,478 +1289,485 @@ Update_History_Loop:
 		sizeChecker.completedID = completedID
 		sizeChecker.msBuilder = msBuilder
 
-	Process_Decision_Loop:
-		for _, d := range request.Decisions {
-			switch *d.DecisionType {
-			case workflow.DecisionTypeScheduleActivityTask:
-				e.metricsClient.IncCounter(metrics.HistoryRespondDecisionTaskCompletedScope,
-					metrics.DecisionTypeScheduleActivityCounter)
-				targetDomainID := domainID
-				attributes := d.ScheduleActivityTaskDecisionAttributes
-				// First check if we need to use a different target domain to schedule activity
-				if attributes.Domain != nil {
-					// TODO: Error handling for ActivitySchedule failed when domain lookup fails
-					domainEntry, err := e.shard.GetDomainCache().GetDomain(*attributes.Domain)
-					if err != nil {
-						return nil, &workflow.InternalServiceError{Message: "Unable to schedule activity across domain."}
+		binChecksum := request.GetBinaryChecksum()
```

Comment:
what happens if no checksum is included in the request?

Reply:
GetBinaryChecksum will return "" and then won't match any key in the map.
I will add a unit test